dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/JdbcEventStore.java

```diff
@@ -1025,6 +1021,15 @@ private StringBuilder getFromWhereClause(
     return fromBuilder;
   }
 
+  private String getOrgUnitSql(EventSearchParams params, String ouTable) {
+    return switch (params.getOrgUnitSelectionMode()) {
+      case SELECTED -> getSelectedOrgUnitPath(params.getAccessibleOrgUnits(), ouTable);
```

Comment:
your TrackedEntityQueryParams javadocs say

```
   * Prepares the organisation units of the given parameters to simplify querying. Mode ACCESSIBLE
is converted to DESCENDANTS for organisation units linked to the search scope of the given
   * user. Mode CAPTURE is converted to DESCENDANTS too, but using organisation units linked to the
   * user's capture scope, and mode CHILDREN is converted to SELECTED for organisation units
   * including all their children. Mode can be DESCENDANTS, SELECTED, ALL only after invoking this
   * method.
```

why are we dealing with modes other than `DESCENDANTS, SELECTED, ALL` here?

Reply:
The javadocs refer to the `/trackedEntities` endpoint, and the `getOrgUnitSql` is meant for events only. I did that change in the docs while I was analyzing the ACL logic, maybe I should have done it in a separate ticket, but since it was such a small change...